src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  let day = '';
+  let month = '';
+  let year = '';
+  const res = [];
+
+  for (let i = 0; i < dateArr.length; i++) {
+    switch (fromFormat[i]) {
+      case 'YY': {
+        year = dateArr[i] < 30 ? `20${dateArr[i]}` : `19${dateArr[i]}`;
+        break;
+      }
+
+      case 'YYYY': {
+        year = dateArr[i];
+        break;
+      }
+
+      case 'DD': {
+        day = dateArr[i];
+        break;
+      }
+
+      case 'MM': {
+        month = dateArr[i];
+        break;
+      }
+    }
+  }
+
+  for (const i of toFormat) {
+    switch (i) {
+      case 'DD': {
+        res.push(day);
+        break;
+      }
+
+      case 'MM': {
+        res.push(month);
+        break;
+      }
+
+      case 'YY': {
+        res.push(year.slice(2));
+        break;
+      }
+
+      case 'YYYY': {
+        res.push(year);
+        break;
+      }
+    }
+  }
+
+  return res.join(toFormat[3]);
```

Comment:
Similar to the issue in line 11, the code assumes that the delimiter is at the same index in the `toFormat` array. This could lead to incorrect results if the delimiter position differs. Consider using a more flexible approach to determine the delimiter.